File: 프로그래머스/0/181931. 등차수열의 특정한 항만 더하기/등차수열의 특정한 항만 더하기.py
def solution(a, d, included):
    arr = [a]
    
    for i in range(len(included)-1):
        
        a = a+d
        arr.append(a)
    sum = 0    
    for i in range(len(included)):
        if included[i] == True:
            sum += arr[i]
    
    return sum

File: 프로그래머스/0/181931. 등차수열의 특정한 항만 더하기/등차수열의 특정한 항만 더하기.java
class Solution {
    public int solution(int a, int d, boolean[] included) {
        int answer = 0;
        
        int startNum = a;
        
        int[] arr = new int[included.length];
        
        for(int i=0; i<included.length; i++) {
            arr[i] = startNum;
            startNum += d;
        }
        
        for(int i=0; i<included.length; i++) {
            if(included[i]) {
                answer += arr[i];
            }
        }
        
        return answer;
    }
}